File: String-2/src/endOther/EndOther.java
package endOther;

public class EndOther {
    //Retorne true se uma das Strings dada estiver contida no final da outra. O programa não deverá fazer a distinção entre
    //maiúsculas e minúsculas, o que deverá importar é a informação da String.

    public boolean endOther(String a, String b) {
        String aInfo = a.toLowerCase(), bInfo = b.toLowerCase();
        return aInfo.endsWith(bInfo) || bInfo.endsWith(aInfo);
    }
}

File: Python/String-2/end_other.py
"""
Dada 2 Strings ("a" e "b"), retorne true se uma delas aparecer no fim da outra.
O programa não deve ser "case sensitive" (ter diferença entre maiúsculo e minúsculo).
Ex.:(('Hiabc', 'abc') → True; ('AbC', 'HiaBc') → True; ('abc', 'abXabc') → True).
"""


def end_other(a, b):
    menor = a.lower() if len(a) < len(b) else b.lower()
    maior = b.lower() if a == menor else a.lower()
    return maior.endswith(menor)


# OUTRO JEITO DE SE FAZER
def end_other_2(a, b):
    a = a.lower()
    b = b.lower()
    return b.endswith(a) or a.endswith(b)


print(end_other('abcXYZ', 'abcDEF'))
